bittensor/_neuron/text/template_miner/__init__.py

```diff
@@ -1,161 +1,702 @@
+    @classmethod
     def check_config( cls, config: 'bittensor.Config' ):
         r""" Checks/validates the config namespace object.
         """
-        assert config.neuron.batch_size_train > 0, "batch_size_train must be a positive value"
-        assert config.neuron.learning_rate > 0, "learning_rate must be a positive value."
+        nucleus.check_config( config )
         bittensor.logging.check_config( config )
         bittensor.wallet.check_config( config )
         bittensor.subtensor.check_config( config )
         bittensor.metagraph.check_config( config )
         bittensor.dataset.check_config( config )
         bittensor.dendrite.check_config( config )
-        bittensor.axon.check_config( config )
         bittensor.wandb.check_config( config )
+        bittensor.axon.check_config( config )
         full_path = os.path.expanduser('{}/{}/{}/{}'.format( config.logging.logging_dir, config.wallet.name, config.wallet.hotkey, config.neuron.name ))
         config.neuron.full_path = os.path.expanduser(full_path)
+        config.using_wandb = config.wandb.api_key != 'default'
         if not os.path.exists(config.neuron.full_path):
             os.makedirs(config.neuron.full_path)
+
+    # ---- Axon Forward call ----
+    def forward_text ( self, inputs_x: torch.FloatTensor) -> torch.FloatTensor:
+        r""" Subscribed to an axon servicing endpoint: processes forward messages from the wire.
+            The arguments reflect an RPC request from another miner in the network, the response tensor
+            should be the hidden units computed using the local context and with shape: [batch_size, sequence_len, __network_dim__].
+
+            Args:
+                inputs_x ( :obj:`torch.Tensor`, `required`):
+                    torch inputs to be forward processed.
+
+            Returns:
+                outputs (:obj:`torch.FloatTensor`):
+                    The nucleus's outputs as a torch tensor of shape [batch_size, sequence_len, __network_dim__]
+        """
+        outputs_y = self.nucleus (
+            inputs_x.to( self.device ),
+            self.metagraph, 
+            self.dendrite, 
+            is_inference = True
+        ).to( self.device )
+        return outputs_y
+
+    # ---- Axon Backward call ----
+    def backward_text ( self, inputs_x:torch.FloatTensor, grads_dy:torch.FloatTensor ):
+        r""" Subscribed to an axon servicing endpoint: Processes backward messages from the wire.
+            Arguments reflect an RPC backward request from another miner in the network. No response
+            needed for tokenized text inputs (uint64s have no gradient).
+
+            Args:
+                inputs_x ( :obj:`torch.Tensor`, `required`):
+                    torch inputs from previous forward call.
+                grads_dy ( :obj:`torch.Tensor`, `required`):
+                    torch grads of forward output.                    
+        """
+        if self.config.neuron.accumulate_remote_gradients:
```

Comment:
accumulate_remote_gradients is not part of the config

Reply:
I am also fairly certain that this old backward function does not work. It might be better to remove it and have the template miner only respond to forward requests

Reply:
Can we make this fix and allow gradients rather than remove
should simply require the config item